From 9363b156829b9aca52b21ca95084334e9f38d691 Mon Sep 17 00:00:00 2001
Message-ID: <9363b156829b9aca52b21ca95084334e9f38d691.1699318202.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1699318202.git.sharlatanus@gmail.com>
References: <cover.1699318202.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Sat, 4 Nov 2023 19:41:43 +0000
Subject: [PATCH 04/16] gnu: cl-markdown: Enable tests.

* gnu/packages/lisp-xyz.scm (sbcl-cl-markdown): Enable tests.
[source]: Update URL as maintainer and owner of repository was changed.
[native-inputs]: Add sbcl-lift and sbcl-trivial-shell.
(ecl-cl-markdown): Keep tests disabled.

Change-Id: Ib27a650ca25d1e90c407411f61b3413443db1f15
---
 gnu/packages/lisp-xyz.scm | 17 +++++++++++------
 1 file changed, 11 insertions(+), 6 deletions(-)

diff --git a/gnu/packages/lisp-xyz.scm b/gnu/packages/lisp-xyz.scm
index 9b47553b14..78f9240f4d 100644
--- a/gnu/packages/lisp-xyz.scm
+++ b/gnu/packages/lisp-xyz.scm
@@ -17740,12 +17740,14 @@ (define-public sbcl-cl-markdown
        (origin
          (method git-fetch)
          (uri (git-reference
-               (url "https://github.com/gwkkwg/cl-markdown")
+               (url "https://github.com/hraban/cl-markdown")
                (commit commit)))
          (file-name (git-file-name "cl-markdown" version))
          (sha256
           (base32 "1algqwmafipyf194cx9wfhg0pdx1ppx6s444p1pm8yaycbsyla1v"))))
       (build-system asdf-build-system/sbcl)
+      (native-inputs
+       (list sbcl-lift sbcl-trivial-shell))
       (inputs
        `(("anaphora" ,sbcl-anaphora)
          ("cl-containers" ,sbcl-cl-containers)
@@ -17753,10 +17755,6 @@ (define-public sbcl-cl-markdown
          ("dynamic-classes" ,sbcl-dynamic-classes)
          ("metabang-bind" ,sbcl-metabang-bind)
          ("metatilities-base" ,sbcl-metatilities-base)))
-      (arguments
-       ;; NOTE: (Sharlatan-20210107213629+0000) Tests depend on too many not
-       ;; available systems, which  themself are abandoned.
-       `(#:tests? #f))
       (home-page "https://common-lisp.net/project/cl-markdown/")
       (synopsis "Common Lisp rewrite of Markdown")
       (description
@@ -17764,7 +17762,14 @@ (define-public sbcl-cl-markdown
       (license license:expat))))
 
 (define-public ecl-cl-markdown
-  (sbcl-package->ecl-package sbcl-cl-markdown))
+  (let ((pkg (sbcl-package->ecl-package sbcl-cl-markdown)))
+    (package
+      (inherit pkg)
+      (arguments
+       ;; XXX: Tests fail with "The function LIFT::GET-BACKTRACE-AS-STRING is
+       ;; undefined" on ECL.
+       ;; See https://github.com/hraban/cl-markdown/issues/11
+       '(#:tests? #f)))))
 
 (define-public cl-markdown
   (sbcl-package->cl-source-package sbcl-cl-markdown))
-- 
2.41.0

